### ISSUE TITLE
cannot find -ltrace

### ISSUE THREAD

**Issue by stsp** [h4ck3rm1k3/traces#4]
/bin/sh ../libtool  --tag=CC   --mode=link gcc  -g -O2 -L../  -o tracedumper filesystem.o trace_dumper.o trace_user_stubs.o -ltrace 
libtool: link: gcc -g -O2 -o tracedumper filesystem.o trace_dumper.o trace_user_stubs.o  -L../ -ltrace
/usr/bin/ld: cannot find -ltrace
collect2: error: ld returned 1 exit status

I am not sure what is -ltrace. Am I missing some
package? If so, configure should have warned me
beforehand.

**Comment by h4ck3rm1k3** [comment]
yes, I can confirm this bug. I was able to reproduce it with 3.5

```
./configure --with-llvm-version=3.5 --with-llvm-prefix=/usr/
```

jamesmikedupont@instance-3:~/traces$ make

```
make  all-recursive
make[1]: Entering directory '/home/jamesmikedupont/traces'
Making all in trace_dumper
make[2]: Entering directory '/home/jamesmikedupont/traces/trace_dumper'
gcc -DHAVE_CONFIG_H -I. -I..  -I ../include   -g -O2 -MT filesystem.o -MD -MP -MF .deps/filesystem.Tpo -c -o filesystem.o filesystem.c
mv -f .deps/filesystem.Tpo .deps/filesystem.Po
gcc -DHAVE_CONFIG_H -I. -I..  -I ../include   -g -O2 -MT trace_dumper.o -MD -MP -MF .deps/trace_dumper.Tpo -c -o trace_dumper.o trace_dumper.c
mv -f .deps/trace_dumper.Tpo .deps/trace_dumper.Po
gcc -DHAVE_CONFIG_H -I. -I..  -I ../include   -g -O2 -MT trace_user_stubs.o -MD -MP -MF .deps/trace_user_stubs.Tpo -c -o trace_user_stubs.o trace_user_stubs.c
mv -f .deps/trace_user_stubs.Tpo .deps/trace_user_stubs.Po
/bin/bash ../libtool  --tag=CC   --mode=link gcc  -g -O2 -L../  -o tracedumper filesystem.o trace_dumper.o trace_user_stubs.o -ltrace
libtool: link: gcc -g -O2 -o tracedumper filesystem.o trace_dumper.o trace_user_stubs.o  -L../ -ltrace
/usr/bin/ld: cannot find -ltrace
collect2: error: ld returned 1 exit status
Makefile:382: recipe for target 'tracedumper' failed
make[2]: *** [tracedumper] Error 1
make[2]: Leaving directory '/home/jamesmikedupont/traces/trace_dumper'
Makefile:395: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/jamesmikedupont/traces'
Makefile:326: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by stsp** [comment]
The situation is very strange.
All I need is to log all the function calls and return
values in the program. The task is very simple, but
a few programs that pretend to do that, all do not compile.
Maybe you know some other project that can help
me with that simple task, and maybe it even compiles. :)

**Comment by h4ck3rm1k3** [comment]
I tried to get traces to work, could not. This was my attempt. I am working in the direction you are talking about but do not have anything. I would suggest maybe to use a combo of gperf, valgrind, strace and ltrace. Can you split out the things you are interested into a shared object and use ltrace?
wait there is one thing you can do http://stackoverflow.com/questions/13961774/gnu-gcc-ld-wrapping-a-call-to-symbol-with-caller-and-callee-defined-in-the-sam you can use the linker to wrap functions.

**Comment by stsp** [comment]
The task is to just find what function returns an
error in a quite big program. So I don't think I can
effectively isolate such functions in advance, or
wrap them: so far I need to trace _all_ functions,
not a subset of them. The projects that use
-finstrument-functions come closer but they do
not support reporting of the return value because
the instrumentation does not allow this.

**Comment by h4ck3rm1k3** [comment]
well I am sorry, I dont have a ready made solution for you, but I
sympathize with you.

On Tue, Apr 5, 2016 at 8:26 AM, Stas Sergeev notifications@github.com
wrote:

> The task is to just find what function returns an
> error in a quite big program. So I don't think I can
> effectively isolate such functions in advance, or
> wrap them: so far I need to trace _all_ functions,
> not a subset of them. The projects that use
> -finstrument-functions come closer but they do
> not support reporting of the return value because
> the instrumentation does not allow this.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> https://github.com/h4ck3rm1k3/traces/issues/4#issuecomment-205778582

## 

James Michael DuPont
Kansas Linux Fest http://kansaslinuxfest.us
Free/Libre Open Source and Open Knowledge Association of Kansas
http://openkansas.us
Member of Free Libre Open Source Software Kosova http://www.flossk.org
Saving Wikipedia(tm) articles from deletion http://SpeedyDeletion.wikia.com
